data_collection/gazette/spiders/rj_sao_joao_de_meriti.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+import json
+import re
+
+import scrapy
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSaoJoaoDeMeritiSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3305109"
+    name = "rj_sao_joao_de_meriti"
+    allowed_domains = ["meriti.rj.gov.br"]
+
+    URL_BEFORE_2018 = "https://meriti.rj.gov.br/dom_mp/"
+    URL_FROM_2018 = (
+        "https://transparencia.meriti.rj.gov.br/diario_oficial_get.php?mesano="
+    )
+
+    start_date = datetime.date(2011, 9, 30)
+    end_date = datetime.date.today()
+
+    # start_date = datetime.date(2017, 12, 25)
+    # end_date = datetime.date(2018, 1, 5)
+
+    def start_requests(self):
+        janeiro_2018 = datetime.date(2018, 1, 1)
+        for month in self.months_interval():
+            if month < janeiro_2018:
+                yield scrapy.Request(
+                    f"{self.URL_BEFORE_2018}{month.year}{month.month:02}.php"
+                )
+            else:
+                yield scrapy.Request(
+                    f"{self.URL_FROM_2018}{month.month}/{month.year}"
+                )
+
+    def parse(self, response):
+        gazettes = []
+
+        if self.URL_BEFORE_2018 in response.url:
+            gazettes = self.parse_month_before_2018(response)
+        else:
+            gazettes = self.parse_month_from_2018(response)
+
+        for gazette in gazettes:
+            yield gazette
+
+    def months_interval(self):
+        month_rule = rrule(freq=MONTHLY, dtstart=self.start_date, until=self.end_date)
+        months = [month.date() for month in month_rule]
+        if months and months[-1].month != self.end_date.month:
+            months.append(self.end_date)
+        return months

```

Comment:
Sorry for my dumb question. But why do you need this `if`? 

```
>>> [month.date() for month in rrule(freq=MONTHLY, dtstart=datetime.date(2020, 1, 1), until=datetime.date.today())]
[datetime.date(2020, 1, 1), datetime.date(2020, 2, 1), datetime.date(2020, 3, 1), datetime.date(2020, 4, 1), datetime.date(2020, 5, 1), datetime.date(2020, 6, 1), datetime.date(2020, 7, 1), datetime.date(2020, 8, 1), datetime.date(2020, 9, 1), datetime.date(2020, 10, 1), datetime.date(2020, 11, 1)]
```

Reply:
If not specified on CLI (-a parameter) the gazette will use the start_date = datetime.date(2011, 9, 30) [line 22]. The rrule will not include the current month if the today() is smaller than 30:

`[month.date() for month in rrule(freq=MONTHLY, dtstart=datetime.date(2011, 9, 30), until=datetime.date.today())][-3:]
`

Reply:
So if someone define (with -a) start_date on day X of some month and today().day is smaller than X, the current month gazettes will be ignored.